pkg/controller/logstorage/secrets/secret_controller.go

```diff
@@ -238,70 +244,71 @@ func (r *SecretSubController) Reconcile(ctx context.Context, request reconcile.R
+		elasticKeys, err := r.generateInternalElasticSecrets(reqLogger, kibanaEnabled, operatorSigner)
 		if err != nil {
-			r.status.SetDegraded(operatorv1.ResourceReadError, "Error building certificate manager", err, reqLogger)
 			return reconcile.Result{}, err
 		}
-		appCM.AddToStatusManager(r.status, render.ElasticsearchNamespace)
-	}
-
-	// Provision secrets and the trusted bundle into the cluster.
-	hdler := utils.NewComponentHandler(reqLogger, r.client, r.scheme, ls)
-
-	// Create Elasticsearch secrets.
-	esTrustedBundle := elasticKeys.trustedBundle(clusterCM)
-	if err = hdler.CreateOrUpdateOrDelete(ctx, elasticKeys.component(esTrustedBundle), r.status); err != nil {
-		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error creating / updating resource", err, reqLogger)
-		return reconcile.Result{}, err
-	}
 
-	if kibanaEnabled {
-		// Render the key pair and trusted bundle into the Kibana namespace.
-		if err = hdler.CreateOrUpdateOrDelete(ctx, elasticKeys.kibanaComponent(esTrustedBundle), r.status); err != nil {
+		// Create Elasticsearch keypair into the tigera-elasticsearch Namespace.
+		if err = hdler.CreateOrUpdateOrDelete(ctx, elasticKeys.internalESComponent(), r.status); err != nil {
 			r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error creating / updating resource", err, reqLogger)
 			return reconcile.Result{}, err
 		}
-	}
 
-	// Generate keys for Tigera components if we're running in multi-tenant mode and this is a reconcile
-	// for a particular tenant, or if not in multi-tenant mode.
-	reconcileTigeraSecrets := !r.multiTenant || r.multiTenant && tenant != nil
-	if !reconcileTigeraSecrets {
-		reqLogger.Info("Skipping render of secrets for Tigera ES components")
-		return reconcile.Result{}, nil
+		if kibanaEnabled {
+			// Render the key pair and trusted bundle into the Kibana namespace.
+			if err = hdler.CreateOrUpdateOrDelete(ctx, elasticKeys.internalKibanaComponent(elasticKeys.trustedBundle(operatorSigner)), r.status); err != nil {
+				r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error creating / updating resource", err, reqLogger)
+				return reconcile.Result{}, err
+			}
+		}
 	}
 	reqLogger.Info("Rendering secrets for Tigera ES components")
 
+	// For single-tenant systems, we use the same root CA and thus the same certificate manager to sign all secrets.
+	cm = operatorSigner
+	if r.multiTenant {
+		// Override with a tenant-scoped certificate manager which uses the CA in the tenant's namespace.
+		opts := []certificatemanager.Option{certificatemanager.WithLogger(reqLogger), certificatemanager.WithTenant(tenant)}
+		cm, err = certificatemanager.Create(r.client, install, r.clusterDomain, helper.InstallNamespace(), opts...)
+		if err != nil {
+			r.status.SetDegraded(operatorv1.ResourceReadError, "Error building certificate manager", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+		cm.AddToStatusManager(r.status, render.ElasticsearchNamespace)
```

Comment:
Is the namespace correct here ?

Reply:
It's effectively ignored, but I'll change it for clarity.